### ISSUE TITLE
What is the difference between replace_prob and mask_prob

### ISSUE THREAD

**Issue by SenYan1999** [lucidrains/mlm-pytorch#1]
Thank you to give such a wonderful tool, and there is a question I want to ask. What is the different between replace_prob and mask_prob? Is a token will be masked just at the probability replace_prob * mask_prob?
``` python
    def __init__(
        self,
        transformer,
        mask_prob = 0.15,
        replace_prob = 0.9,
        num_tokens = None,
        random_token_prob = 0.,
        mask_token_id = 2,
        pad_token_id = 0,
        mask_ignore_token_ids = []):
```

**Comment by lucidrains** [comment]
@SenYan1999 that's exactly right!

![dog is hairy](https://user-images.githubusercontent.com/108653/100179959-f6bf5c00-2e8b-11eb-9f12-67263bc4c233.png)

this is necessary as specified in the paper, so that the attention net learns how to generate sound logits for unmasked tokens
